Support Stargate with 4.0.0-beta4

**Is your feature request related to a problem? Please describe.**
With 4.0.0 support now in place, stargate deployment needs to be updated to accommodate.

Currently, when you deploy stargate with a 4.0.0 cluster stargate will fail to start with errors like:

```
io.stargate.core.activator.ServiceStartException: Unable to start persistence-cassandra-3.11
Detected service startup failure in bundle io.stargate.db.cassandra: io.stargate.core.activator.ServiceStartException: Unable to start persistence-cassandra-3.11
        at io.stargate.core.activator.BaseActivator.startServiceInternal(BaseActivator.java:182)
        at io.stargate.core.activator.BaseActivator.access$900(BaseActivator.java:37)
        at io.stargate.core.activator.BaseActivator$Tracker.startIfAllRegistered(BaseActivator.java:225)
        at io.stargate.core.activator.BaseActivator$Tracker.addingService(BaseActivator.java:201)
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:943)
        at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:871)
        at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
        at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
        at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:903)
        at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990)
        at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838)
        at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545)
        at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4833)
        at org.apache.felix.framework.Felix.registerService(Felix.java:3804)
        at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:328)
        at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:302)
        at io.stargate.core.activator.BaseActivator.startServiceInternal(BaseActivator.java:175)
        at io.stargate.core.activator.BaseActivator.start(BaseActivator.java:98)
        at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:698)
        at org.apache.felix.framework.Felix.activateBundle(Felix.java:2402)
        at org.apache.felix.framework.Felix.startBundle(Felix.java:2308)
        at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:998)
        at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:984)
        at io.stargate.starter.Starter.start(Starter.java:431)
        at io.stargate.starter.Starter.cli(Starter.java:607)
        at io.stargate.starter.Starter.main(Starter.java:648)
Caused by: java.lang.RuntimeException: A node with address /10.244.2.12 already exists, cancelling join. Use cassandra.replace_address if you want to replace this node.
        at org.apache.cassandra.service.StorageService.checkForEndpointCollision(StorageService.java:597)
        at org.apache.cassandra.service.StorageService.prepareToJoin(StorageService.java:844)
        at org.apache.cassandra.service.StorageService.initServer(StorageService.java:703)
        at org.apache.cassandra.service.StorageService.initServer(StorageService.java:652)
        at org.apache.cassandra.service.CassandraDaemon.setup(CassandraDaemon.java:397)
        at org.apache.cassandra.service.CassandraDaemon.init(CassandraDaemon.java:521)
        at io.stargate.db.cassandra.impl.CassandraPersistence.initializePersistence(CassandraPersistence.java:171)
        at io.stargate.db.cassandra.impl.CassandraPersistence.initializePersistence(CassandraPersistence.java:94)
        at io.stargate.db.datastore.common.AbstractCassandraPersistence.initialize(AbstractCassandraPersistence.java:106)
        at io.stargate.db.cassandra.CassandraPersistenceActivator.createService(CassandraPersistenceActivator.java:169)
        at io.stargate.core.activator.BaseActivator.createServices(BaseActivator.java:238)
        at io.stargate.core.activator.BaseActivator.startServiceInternal(BaseActivator.java:170)
        ... 25 more
INFO  [main] 2021-02-20 13:38:38,657 BaseActivator.java:173 - Registering core services as com.codahale.metrics.health.HealthCheckRegistry
INFO  [main] 2021-02-20 13:38:38,657 BaseActivator.java:180 - Started core services
Starting bundle io.stargate.cql
INFO  [main] 2021-02-20 13:38:38,953 BaseActivator.java:92 - Starting CQL ...
Starting bundle io.stargate.graphql
INFO  [main] 2021-02-20 13:38:39,393 BaseActivator.java:92 - Starting GraphQL ...
Starting bundle io.stargate.health
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [bundle://ba6c254a-dc3f-4246-b77d-bc04db9f2d36_20.0:82/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [bundle://ba6c254a-dc3f-4246-b77d-bc04db9f2d36_20.0:100/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
INFO  [main] 2021-02-20 13:38:39,792 BaseActivator.java:92 - Starting healthchecker ...
INFO  [main] 2021-02-20 13:38:39,797 HealthCheckerActivator.java:37 - Starting healthchecker....
INFO  [2021-02-20 13:38:43,310] io.dropwizard.server.SimpleServerFactory: Registering jersey handler with root path prefix: /checker/
WARN  [2021-02-20 13:38:43,322] org.eclipse.jetty.server.handler.ContextHandler: i.d.j.MutableServletContextHandler@730bb6{/,null,UNAVAILABLE} contextPath ends with /
INFO  [2021-02-20 13:38:43,326] io.dropwizard.server.SimpleServerFactory: Registering admin handler with root path prefix: /
INFO  [2021-02-20 13:38:43,348] io.dropwizard.server.SimpleServerFactory: Registering jersey handler with root path prefix: /checker/
WARN  [2021-02-20 13:38:43,348] org.eclipse.jetty.server.handler.ContextHandler: i.d.j.MutableServletContextHandler@730bb6{/checker,null,UNAVAILABLE} contextPath ends with /
INFO  [2021-02-20 13:38:43,348] io.dropwizard.server.SimpleServerFactory: Registering admin handler with root path prefix: /
INFO  [2021-02-20 13:38:43,349] io.dropwizard.server.ServerFactory: Starting Server
INFO  [2021-02-20 13:38:43,793] org.eclipse.jetty.setuid.SetUIDListener: Opened Server@34846a1a{HTTP/1.1, (http/1.1)}{0.0.0.0:8084}
INFO  [2021-02-20 13:38:43,800] org.eclipse.jetty.server.Server: jetty-9.4.30.v20200611; built: 2020-06-11T12:34:51.929Z; git: 271836e4c1f4612f12b7bb13ef5a92a927634b0d; jvm 1.8.0_252-b09
INFO  [2021-02-20 13:38:48,520] io.dropwizard.jersey.DropwizardResourceConfig: The following paths were found for the configured resources:

    GET     /liveness (io.stargate.health.CheckerResource)
    GET     /readiness (io.stargate.health.CheckerResource)

INFO  [2021-02-20 13:38:48,539] org.eclipse.jetty.server.handler.ContextHandler: Started i.d.j.MutableServletContextHandler@730bb6{/checker,null,AVAILABLE}
INFO  [2021-02-20 13:38:48,563] io.dropwizard.setup.AdminEnvironment: tasks = 

    POST    /tasks/log-level (io.dropwizard.servlets.tasks.LogConfigurationTask)
    POST    /tasks/gc (io.dropwizard.servlets.tasks.GarbageCollectionTask)

INFO  [2021-02-20 13:38:48,704] org.eclipse.jetty.server.handler.ContextHandler: Started i.d.j.MutableServletContextHandler@d97abe7{/,null,AVAILABLE}
INFO  [2021-02-20 13:38:48,891] org.eclipse.jetty.server.AbstractConnector: Started Server@34846a1a{HTTP/1.1, (http/1.1)}{0.0.0.0:8084}
INFO  [2021-02-20 13:38:48,891] org.eclipse.jetty.server.Server: Started @22772ms
INFO  [2021-02-20 13:38:48,899] io.stargate.health.HealthCheckerActivator: Started healthchecker....
INFO  [main] 2021-02-20 13:38:48,899 BaseActivator.java:180 - Started healthchecker
Starting bundle org.apache.felix.scr
Starting bundle null
Starting bundle null
Starting bundle osgi.cmpn
Starting bundle org.osgi.util.function
Starting bundle org.osgi.util.promise
Starting bundle io.stargate.db
INFO  [main] 2021-02-20 13:38:49,457 BaseActivator.java:92 - Starting DB services ...
Starting bundle io.stargate.db.limiter.global
INFO  [main] 2021-02-20 13:38:49,523 BaseActivator.java:92 - Starting Global Rate Limiting ...
INFO  [main] 2021-02-20 13:38:49,523 BaseActivator.java:180 - Started Global Rate Limiting
Starting bundle io.stargate.web
10.244.1.4 - - [20/Feb/2021:13:38:52 +0000] "GET /metrics HTTP/1.1" 200 1662833 "-" "Prometheus/2.22.1" 2481
INFO  [main] 2021-02-20 13:38:52,390 BaseActivator.java:92 - Starting restapi ...
Terminating due to previous service startup errors.
INFO  [StorageServiceShutdownHook] 2021-02-20 13:38:52,441 HintsService.java:209 - Paused hints dispatch
WARN  [StorageServiceShutdownHook] 2021-02-20 13:38:52,442 Gossiper.java:1655 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
INFO  [StorageServiceShutdownHook] 2021-02-20 13:38:52,444 MessagingService.java:985 - Waiting for messaging service to quiesce
INFO  [ACCEPT-/10.244.2.12] 2021-02-20 13:38:52,450 MessagingService.java:1346 - MessagingService has terminated the accept() thread
INFO  [2021-02-20 13:38:52,537] org.eclipse.jetty.server.AbstractConnector: Stopped Server@34846a1a{HTTP/1.1, (http/1.1)}{0.0.0.0:8084}
INFO  [2021-02-20 13:38:52,596] org.eclipse.jetty.server.handler.ContextHandler: Stopped i.d.j.MutableServletContextHandler@d97abe7{/,null,UNAVAILABLE}
INFO  [2021-02-20 13:38:52,879] org.eclipse.jetty.server.handler.ContextHandler: Stopped i.d.j.MutableServletContextHandler@730bb6{/checker,null,UNAVAILABLE}
INFO  [StorageServiceShutdownHook] 2021-02-20 13:38:53,767 HintsService.java:209 - Paused hints dispatch
```

**Describe the solution you'd like**
Stargate should function properly when running a 4.0.0 cluster.

**Additional context**
We knew we would need to do this, but adding a specific issue to make sure it doesn't fall through the cracks.

While we do this we should probably bump to the latest Stargate release available as well.

--- a/charts/k8ssandra/Chart.yaml
+++ b/charts/k8ssandra/Chart.yaml
@@ -3,7 +3,7 @@ name: k8ssandra
 description: |
   Provisions and configures an instance of the entire K8ssandra stack. This includes Apache Cassandra, Stargate, Reaper, Medusa, Prometheus, and Grafana.
 type: application
-version: 0.53.0
+version: 0.54.0
 appVersion: 3.11.10
 
 dependencies:

--- a/charts/k8ssandra/templates/stargate/stargate.yaml
+++ b/charts/k8ssandra/templates/stargate/stargate.yaml
@@ -1,6 +1,13 @@
 {{- if .Values.stargate.enabled }}
-{{- $heapMB := required "stargate.heapMB is required" .Values.stargate.heapMB -}}
-{{- $image := default "stargateio/stargate-3_11:v1.0.7" .Values.stargate.image -}}
+  {{- $stargateVersion := "1.0.8" }}
+  {{- $heapMB := required "stargate.heapMB is required" .Values.stargate.heapMB -}}
+  {{- $clusterVersion := "3.11" }}
+  {{- $computedImage := printf "stargateio/stargate-3_11:v%s" $stargateVersion }}
+  {{- if eq .Values.cassandra.version "4.0.0" }}
+    {{ $clusterVersion = "4.0" }}
+    {{- $computedImage = printf "stargateio/stargate-4_0:v%s" $stargateVersion }}
+  {{- end }}
+  {{- $image := default $computedImage .Values.stargate.image -}}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
@@ -84,7 +91,7 @@ spec:
             - name: CLUSTER_NAME
               value: {{ include "k8ssandra.clusterName" . | quote }}
             - name: CLUSTER_VERSION
-              value: "3.11"
+              value: {{ $clusterVersion | quote }}
             - name: SEED
               value: "{{ include "k8ssandra.clusterName" . }}-seed-service.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain | default "cluster.local" }}"
             - name: DATACENTER_NAME

--- a/charts/k8ssandra/templates/stargate/stargate.yaml
+++ b/charts/k8ssandra/templates/stargate/stargate.yaml
@@ -3,7 +3,7 @@
   {{- $heapMB := required "stargate.heapMB is required" .Values.stargate.heapMB -}}
   {{- $clusterVersion := "3.11" }}
   {{- $computedImage := printf "stargateio/stargate-3_11:v%s" $stargateVersion }}
-  {{- if eq .Values.cassandra.version "4.0.0" }}
+  {{- if not (hasPrefix "3" .Values.cassandra.version) }}
     {{ $clusterVersion = "4.0" }}
     {{- $computedImage = printf "stargateio/stargate-4_0:v%s" $stargateVersion }}
   {{- end }}

--- a/charts/k8ssandra/templates/stargate/stargate.yaml
+++ b/charts/k8ssandra/templates/stargate/stargate.yaml
@@ -1,13 +1,17 @@
 {{- if .Values.stargate.enabled }}
-  {{- $stargateVersion := "1.0.8" }}
-  {{- $heapMB := required "stargate.heapMB is required" .Values.stargate.heapMB -}}
-  {{- $clusterVersion := "3.11" }}
-  {{- $computedImage := printf "stargateio/stargate-3_11:v%s" $stargateVersion }}
-  {{- if not (hasPrefix "3" .Values.cassandra.version) }}
-    {{ $clusterVersion = "4.0" }}
+  {{- $stargateVersion := .Values.stargate.version }}
+  {{- $computedClusterVersion := "" }}
+  {{- $computedImage := "" }}
+  {{- if hasPrefix "3" .Values.cassandra.version }}
+    {{- $computedClusterVersion = "3.11" }}
+    {{- $computedImage = printf "stargateio/stargate-3_11:v%s" $stargateVersion }}
+  {{- else }}
+    {{- $computedClusterVersion = "4.0" }}
     {{- $computedImage = printf "stargateio/stargate-4_0:v%s" $stargateVersion }}
   {{- end }}
   {{- $image := default $computedImage .Values.stargate.image -}}
+  {{- $clusterVersion := default $computedClusterVersion .Values.stargate.clusterVersion -}}
+  {{- $heapMB := required "stargate.heapMB is required" .Values.stargate.heapMB -}}
 apiVersion: apps/v1
 kind: Deployment
 metadata:

--- a/charts/k8ssandra/values.yaml
+++ b/charts/k8ssandra/values.yaml
@@ -167,17 +167,26 @@ stargate:
   # -- Enable Stargate resources as part of this release
   enabled: true
 
+  # -- version of Stargate to deploy
+  version: "1.0.8"
+
   # -- Number of Stargate instances to deploy. This value may be scaled independently of
   # Cassandra cluster nodes. Each instance handles API and coordination tasks
   # for inbound queries.
   replicas: 1
 
-  # -- Sets the Stargate container image. If left blank (recommended), k8ssandra
-  # will derive an appropriate image based on your cluster version.
+  # -- Sets the Stargate container image. This value must be compatible
+  # with the value provided for stargate.clusterVersion. If left blank (recommended),
+  # k8ssandra will derive an appropriate image based on cassandra.clusterVersion.
   image:
 
   # -- Sets the imagePullPolicy used by the Stargate pods
   imagePullPolicy: IfNotPresent
+
+  # -- Sets the CLUSTER_VERSION environment variable provided to the Stargate containers.
+  # This value must be compatible with the value provided for stargate.image. If left blank (recommended),
+  # k8ssandra will derive an appropriate image based on cassandra.clusterVersion.
+  clusterVersion:
 
   # -- Sets the heap size Stargate will use in megabytes. Memory request and
   # limit for the pod will be set to this value x2 and x4, respectively.

--- a/docs/content/en/docs/faqs/_index.md
+++ b/docs/content/en/docs/faqs/_index.md
@@ -112,7 +112,7 @@ K8ssandra deploys the following components, some components are optional, and de
 * [Cassandra Reaper](http://cassandra-reaper.io/)
   * 2.1.3
 * [Stargate](https://stargate.io/)
-  * 1.0.7
+  * 1.0.8
 
 *Note: Throughout these docs, examples are shown to deploy [Traefik](https://traefik.io/) as a means to provide external access to the k8ssandra cluster.  It is deployed separately from k8ssandra, and as such, the version deployed will vary.*
 

--- a/tests/unit/template_stargate_test.go
+++ b/tests/unit/template_stargate_test.go
@@ -12,6 +12,15 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 )
 
+const (
+	DefaultStargate3Image          = "stargateio/stargate-3_11:v1.0.8"
+	DefaultStargate4Image          = "stargateio/stargate-4_0:v1.0.8"
+	DefaultStargate3ClusterVersion = "3.11"
+	DefaultStargate4ClusterVersion = "4.0"
+	DefaultStargateImage           = DefaultStargate3Image
+	DefaultStargateClusterVersion  = DefaultStargate3ClusterVersion
+)
+
 var _ = Describe("Verify Stargate template", func() {
 	var (
 		helmChartPath string
@@ -90,7 +99,7 @@ var _ = Describe("Verify Stargate template", func() {
 
 			Expect(len(templateSpec.Containers)).To(Equal(1))
 			container := templateSpec.Containers[0]
-			Expect(container.Image).To(Equal("stargateio/stargate-3_11:v1.0.8"))
+			Expect(container.Image).To(Equal(DefaultStargateImage))
 			Expect(container.Name).To(Equal(Sprintf("%s-dc1-stargate", HelmReleaseName)))
 			Expect(string(container.ImagePullPolicy)).To(Equal("IfNotPresent"))
 
@@ -111,13 +120,91 @@ var _ = Describe("Verify Stargate template", func() {
 			Expect(clusterName.Value).To(Equal(HelmReleaseName))
 
 			clusterVersion := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
-			Expect(clusterVersion.Value).To(Equal("3.11"))
+			Expect(clusterVersion.Value).To(Equal(DefaultStargateClusterVersion))
 
 			seed := kubeapi.FindEnvVarByName(container, "SEED")
 			Expect(seed.Value).To(Equal(Sprintf("%s-seed-service.%s.svc.cluster.local", HelmReleaseName, DefaultTestNamespace)))
 
 			datacenterName := kubeapi.FindEnvVarByName(container, "DATACENTER_NAME")
 			Expect(datacenterName.Value).To(Equal("dc1"))
+		})
+
+		It("using custom image and clusterVersion", func() {
+			// This combination of values makes no real sense and would not work
+			// but this tests that the defaults are avoided when a specific value is provided
+			image := "stargateio/stargate-4_0:v1.0.5"
+			clusterVersion := "3.0"
+
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				SetValues: map[string]string{
+					"stargate.enabled":        "true",
+					"stargate.image":          image,
+					"stargate.clusterVersion": clusterVersion,
+				},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			templateSpec := deployment.Spec.Template.Spec
+			Expect(len(templateSpec.Containers)).To(Equal(1))
+			container := templateSpec.Containers[0]
+			Expect(container.Image).To(Equal(image))
+			clusterVersionEnv := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
+			Expect(clusterVersionEnv.Value).To(Equal(clusterVersion))
+		})
+
+		It("using defaults and empty clusterVersion", func() {
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				SetValues: map[string]string{
+					"stargate.enabled":        "true",
+					"stargate.clusterVersion": "",
+				},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			templateSpec := deployment.Spec.Template.Spec
+			Expect(len(templateSpec.Containers)).To(Equal(1))
+			container := templateSpec.Containers[0]
+			Expect(container.Image).To(Equal(DefaultStargateImage))
+			clusterVersionEnv := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
+			Expect(clusterVersionEnv.Value).To(Equal(DefaultStargateClusterVersion))
+		})
+
+		It("using cassandra version 4.0.0", func() {
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				SetValues: map[string]string{
+					"stargate.enabled":  "true",
+					"cassandra.version": "4.0.0",
+				},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			templateSpec := deployment.Spec.Template.Spec
+			Expect(len(templateSpec.Containers)).To(Equal(1))
+			container := templateSpec.Containers[0]
+			Expect(container.Image).To(Equal(DefaultStargate4Image))
+			clusterVersionEnv := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
+			Expect(clusterVersionEnv.Value).To(Equal(DefaultStargate4ClusterVersion))
+		})
+
+		It("using cassandra version 3.11.10", func() {
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				SetValues: map[string]string{
+					"stargate.enabled":  "true",
+					"cassandra.version": "3.11.10",
+				},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			templateSpec := deployment.Spec.Template.Spec
+			Expect(len(templateSpec.Containers)).To(Equal(1))
+			container := templateSpec.Containers[0]
+			Expect(container.Image).To(Equal(DefaultStargate3Image))
+			clusterVersionEnv := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
+			Expect(clusterVersionEnv.Value).To(Equal(DefaultStargate3ClusterVersion))
 		})
 
 		It("changing cluster name", func() {
@@ -221,22 +308,19 @@ var _ = Describe("Verify Stargate template", func() {
 			Expect(string(container.ImagePullPolicy)).To(Equal(alternatePullPolicy))
 		})
 
-		It("using Cassandra 4.0", func() {
+		It("changing stargate version", func() {
+			alternateImage := "stargateio/stargate-3_11:v1.0.5"
 			options := &helm.Options{
 				KubectlOptions: defaultKubeCtlOptions,
 				SetValues: map[string]string{
-					"stargate.enabled":                    "true",
-					"cassandra.version":                   "4.0.0",
-					"cassandra.versionImageMap.4\\.0\\.0": "some/irrelevant:image",
+					"stargate.enabled": "true",
+					"stargate.version": "1.0.5",
 				},
 			}
 
 			Expect(renderTemplate(options)).To(Succeed())
-			Expect(deployment.Spec.Template.Spec.Containers).ToNot(BeNil())
-			Expect(len(deployment.Spec.Template.Spec.Containers)).To(Equal(1))
 			container := deployment.Spec.Template.Spec.Containers[0]
-			clusterVersion := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
-			Expect(clusterVersion.Value).To(Equal("4.0"))
+			Expect(container.Image).To(Equal(alternateImage))
 		})
 	})
 })

--- a/tests/unit/template_stargate_test.go
+++ b/tests/unit/template_stargate_test.go
@@ -90,7 +90,7 @@ var _ = Describe("Verify Stargate template", func() {
 
 			Expect(len(templateSpec.Containers)).To(Equal(1))
 			container := templateSpec.Containers[0]
-			Expect(container.Image).To(Equal("stargateio/stargate-3_11:v1.0.7"))
+			Expect(container.Image).To(Equal("stargateio/stargate-3_11:v1.0.8"))
 			Expect(container.Name).To(Equal(Sprintf("%s-dc1-stargate", HelmReleaseName)))
 			Expect(string(container.ImagePullPolicy)).To(Equal("IfNotPresent"))
 
@@ -109,6 +109,9 @@ var _ = Describe("Verify Stargate template", func() {
 
 			clusterName := kubeapi.FindEnvVarByName(container, "CLUSTER_NAME")
 			Expect(clusterName.Value).To(Equal(HelmReleaseName))
+
+			clusterVersion := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
+			Expect(clusterVersion.Value).To(Equal("3.11"))
 
 			seed := kubeapi.FindEnvVarByName(container, "SEED")
 			Expect(seed.Value).To(Equal(Sprintf("%s-seed-service.%s.svc.cluster.local", HelmReleaseName, DefaultTestNamespace)))
@@ -216,6 +219,24 @@ var _ = Describe("Verify Stargate template", func() {
 			container := deployment.Spec.Template.Spec.Containers[0]
 			Expect(container.Image).To(Equal(alternateImage))
 			Expect(string(container.ImagePullPolicy)).To(Equal(alternatePullPolicy))
+		})
+
+		It("using Cassandra 4.0", func() {
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				SetValues: map[string]string{
+					"stargate.enabled":                    "true",
+					"cassandra.version":                   "4.0.0",
+					"cassandra.versionImageMap.4\\.0\\.0": "some/irrelevant:image",
+				},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			Expect(deployment.Spec.Template.Spec.Containers).ToNot(BeNil())
+			Expect(len(deployment.Spec.Template.Spec.Containers)).To(Equal(1))
+			container := deployment.Spec.Template.Spec.Containers[0]
+			clusterVersion := kubeapi.FindEnvVarByName(container, "CLUSTER_VERSION")
+			Expect(clusterVersion.Value).To(Equal("4.0"))
 		})
 	})
 })

--- a/tests/unit/template_stargate_test.go
+++ b/tests/unit/template_stargate_test.go
@@ -322,5 +322,21 @@ var _ = Describe("Verify Stargate template", func() {
 			container := deployment.Spec.Template.Spec.Containers[0]
 			Expect(container.Image).To(Equal(alternateImage))
 		})
+
+		It("changing stargate version with Cassandra 4.0", func() {
+			alternateImage := "stargateio/stargate-4_0:v1.0.6"
+			options := &helm.Options{
+				KubectlOptions: defaultKubeCtlOptions,
+				SetValues: map[string]string{
+					"stargate.enabled":  "true",
+					"stargate.version":  "1.0.6",
+					"cassandra.version": "4.0.0",
+				},
+			}
+
+			Expect(renderTemplate(options)).To(Succeed())
+			container := deployment.Spec.Template.Spec.Containers[0]
+			Expect(container.Image).To(Equal(alternateImage))
+		})
 	})
 })

